_plugins/generate-api-v1.rb

```diff
@@ -0,0 +1,92 @@
+# This script creates API files for version 1 of the endoflife.date API.
+#
+# There are two kind of generated files :
+# - all.json: contains endoflife metadata, such as the list of all the products and the product count.
+# - <product>.json: contains a given product data (including releases data).
+
+require 'jekyll'
+
+module ApiV1
+
+  class ApiGenerator < Jekyll::Generator
+    safe true
+    priority :lowest
+
+    def generate(site)
+      all_products = []
+
+      site.pages.each do |page|
+        is_markdown = (page.name.end_with?("md"))
+        has_permalink = (page.data.has_key? 'permalink')
+        has_releases = (page.data.has_key? 'releases')
+        is_product_page = (is_markdown and has_permalink and has_releases)
+
+        if is_product_page
```

Comment:
A simpler check would be `page['layout'] == 'product'`

Reply:
You are right, don't know why I made something this complicated.